File: Bridge/Vasylyshyn/src/Shape/Triangle.java
package Shape;

import Color.Color;

public class Triangle extends Shape {
    int SideC;
    int SideA;
    int SideB;

    public Triangle(Color color, int sideC, int sideA, int sideB) {
        super(color);
        SideC = sideC;
        SideA = sideA;
        SideB = sideB;
    }

    public Triangle(int sideC) {
        SideC = sideC;
    }


    public Triangle() {

    }

    public int getSideC() {
        return SideC;
    }

    public Triangle setSideC(int sideC) {
        SideC = sideC;
        return this;
    }

    @Override
    public String toString() {
        return "Triangle{" +
                "SideC=" + SideC +
                ", SideA=" + SideA +
                ", SideB=" + SideB +
                ", color=" + color +
                '}';
    }

    @Override
    public double Area() {
        return Math.sqrt(Perimeter()/2*((Perimeter()/2-this.SideA)*(Perimeter()/2-this.SideB)*(Perimeter()/2-this.SideC)));
    }

    @Override
    public double Perimeter() {
        return this.SideC+this.SideA+this.SideB;
    }

    @Override
    public void draw() {
        System.out.println("It is Triangle. It is already painted");
        color.Paint();

    }

    public int getSideA() {
        return SideA;
    }

    public Triangle setSideA(int sideA) {
        SideA = sideA;
        return this;
    }

    public int getSideB() {
        return SideB;
    }

    public Triangle setSideB(int sideB) {
        SideB = sideB;
        return this;
    }
}

File: Bridge/KovalchukD-03/ConsoleApp25/triangle.cs
using System;

namespace BridgePattern
{
    class Triangle : Shape
    {
        private double sideA;
        private double sideB;
        private double sideC;

        public Triangle(double sideA, double sideB, double sideC, string color) : base(color)
        {
            if (sideA <= 0 || sideB <= 0 || sideC <= 0 || !IsValidTriangle(sideA, sideB, sideC))
            {
                throw new ArgumentException("Invalid sides for a triangle.");
            }

            this.sideA = sideA;
            this.sideB = sideB;
            this.sideC = sideC;
        }

        private bool IsValidTriangle(double a, double b, double c)
        {
            return a + b > c && a + c > b && b + c > a;
        }

        public override double Area()
        {
            if (!IsValidTriangle(sideA, sideB, sideC))
            {
                throw new InvalidOperationException("Cannot calculate area for an invalid triangle.");
            }

            double s = (sideA + sideB + sideC) / 2;

            double area = Math.Sqrt(s * (s - sideA) * (s - sideB) * (s - sideC));

            return Math.Round(area, 3);
        }

        public override double Perimeter()
        {
            return sideA + sideB + sideC;
        }

        public override void Draw()
        {
            Console.WriteLine($"Triangle: Perimeter={Perimeter()}, Area={Area()}, Color={color}");
        }
    }
}
